src/com/akshay/captain/jira.main.kts

```diff
@@ -0,0 +1,240 @@
+    val jira = JiraClient(clueJiraUrl, cred)
+    val releaseNotes = mutableListOf<String>()
+
+    try {
+        val result: Issue.SearchResult = jira.searchIssues(readyForReleaseFilter)
+
+        val tickets = result.generateTickets(platform)
+
+        // 2. Print all changes
+        tickets
+            .filter { it.ticketType.isChange() }
+            .ifNotEmpty { releaseNotes.add(changes) }
+            .map { releaseNotes.add("$changeEmoji ${it.key} \t ${it.summary}") }
+
+        // 3. Print all tech changes
+        tickets
+            .filter { it.ticketType is TicketType.Chapter }
+            .ifNotEmpty { releaseNotes.add(techChanges) }
+            .map { releaseNotes.add("$techChangesEmoji ${it.key} \t ${it.summary}") }
+
+        // 4. Print all bug fixes
+        tickets
+            .filter { it.ticketType is TicketType.Bug }
+            .ifNotEmpty { releaseNotes.add(bugFixes) }
+            .map { releaseNotes.add("$bugEmoji ${it.key} \t ${it.summary}") }
+
+        // 5. Print all instrumentation
+        tickets
+            .filter { it.ticketType is TicketType.Analytics }
+            .ifNotEmpty { releaseNotes.add(instrumentation) }
+            .map { releaseNotes.add("$instrumentationEmoji ${it.key} \t ${it.summary}") }
+
+//        val notes = releaseNotes.toList().joinToString(separator = "\n")
+//        println(notes)
+
+    } catch (exception: Exception) {
+        println(exception.message)
+
+    }
+}
+
+/**
+ * Maps string to a [TicketType]
+ */
+fun IssueType.toTicketType(platform: Platform): TicketType {
+    return when (this.toString()) {
+        "Analytics:${platform}" -> TicketType.Analytics
+        "Story:${platform}" -> TicketType.Story
+        "Sub-Task:${platform}" -> TicketType.SubTask
+        "Improvement:${platform}" -> TicketType.Improvement
+        "Bug:${platform}" -> TicketType.Bug
+        else -> TicketType.Unknown(this.toString())
+    }
+}
+
+/**
+ * Describes type of the platform
+ */
+sealed class Platform {
+    object Android : Platform() {
+        override fun toString(): String = android
+    }
+
+    object iOS : Platform() {
+        override fun toString(): String = ios
+    }
+}
+
+/**
+ * Maps string to a [Platform] type.
+ */
+fun String.toPlatform(): Platform {
+    return when (lowercase()) {
+        android -> Platform.Android
+        ios -> Platform.iOS
+        else -> error("The platform type needs to be either Android or iOS")
+    }
+}
+
+/**
+ * Describes various types of tickets
+ */
+sealed class TicketType {
+    object Analytics : TicketType()
+    object Story : TicketType()
+    object SubTask : TicketType()
+    object Improvement : TicketType()
+    object Bug : TicketType()
+    object Chapter : TicketType()
+    data class Unknown(val type: String) : TicketType()
+
+    /**
+     * Returns if the ticket type is a "change"
+     * which can either be a story, subtask or
+     * an improvement.
+     */
+    fun isChange(): Boolean {
```

Comment:
Suggestion: Rather than a function I would have it as a property but I'm fine either way

```
val isChange = when (this) {
            Story -> true
            SubTask -> true
            Improvement -> true
            else -> false
        }
```

Reply:
It was there before, but yes can be changed